packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,301 @@
+// @flow strict
+import * as React from 'react';
+import { useState, useRef, type Node } from 'react';
+import PropTypes from 'prop-types';
+import TypeaheadInputField from './TypeaheadInputField.js';
+import Option from './TypeaheadOption.js';
+import Box from './Box.js';
+import Text from './Text.js';
+import Flyout from './Flyout.js';
+import Layer from './Layer.js';
+
+type OptionObject = {|
+  label: string,
+  value: string,
+|};
+
+type Props = {|
+  data: Array<{
+    label: string,
+    value: string,
+  }>,
+  defaultItem?: OptionObject,
+  id: string,
+  label: string,
+  noResultText: string,
+  onBlur?: ({
+    event:
+      | SyntheticFocusEvent<HTMLInputElement>
+      | SyntheticEvent<HTMLInputElement>,
+  }) => void,
+  onChange?: ({
+    event: SyntheticInputEvent<HTMLInputElement>,
+    value: string,
+  }) => void,
+  onFocus?: ({
+    event: SyntheticFocusEvent<HTMLInputElement>,
+    value: string,
+  }) => void,
+  onSelect?: (?OptionObject) => void,
```

Comment:
@simkessy could we update the flow type so it reflects the docs? In Gestalt we always return an object in the callback.

`({ event: SyntheticInputEvent<>, value: string }) => void`?

Perhaps it should also include the label?
`({ event: SyntheticInputEvent<>, value: string, label: string }) => void`?